src/qibocal/auto/execute.py

```diff
@@ -121,23 +121,25 @@ def current(self):
         assert self.head is not None
         return self.graph.task(self.head)
 
-    def run(self):
+    def run(self, mode):
         """Actual execution.
 
         The platform's update method is called if:
         - self.update is True and task.update is None
         - task.update is True
         """
         self.head = self.graph.start
-
         while self.head is not None:
             task = self.current
-            task_execution = task.run(platform=self.platform, qubits=self.qubits)
             completed = Completed(task, Normal(), self.output)
-            completed.data = next(task_execution)
-            completed.results = next(task_execution)
+            task_execution = task.run(platform=self.platform, qubits=self.qubits)
+            if mode.name not in ["report"]:
+                completed.data = next(task_execution)
+                if mode.name in ["autocalibration", "fit"]:
+                    completed.results = next(task_execution)
             self.history.push(completed)
             self.head = self.next()
-            if self.platform is not None:
-                if self.update and task.update:
-                    self.platform.update(completed.results.update)
+            if mode.name in ["autocalibration", "fit"]:
+                if self.platform is not None:
+                    if self.update and task.update:
```

Comment:
You can also use `and`:
```suggestion
            if mode.name in ["autocalibration", "fit"] and self.platform is not None and self.update and task.update:
```

Reply:
If you want to group conditions, you can alias them with:
- the benefit of giving them a name
- the drawback of evaluating them (while `and` is short-circuit, like nested `if`s)

```suggestion
            update = self.update and task.update
            if mode.name in ["autocalibration", "fit"] and self.platform is not None and update:
```